ams/server/src/main/java/com/netease/arctic/server/table/executor/TableRuntimeRefreshExecutor.java

```diff
@@ -51,6 +53,17 @@ public void handleStatusChanged(TableRuntime tableRuntime, OptimizingStatus orig
     tryEvaluatingPendingInput(tableRuntime, loadTable(tableRuntime));
   }
 
+  @Override
+  public void handleConfigChanged(TableRuntime tableRuntime, TableConfiguration originalConfig) {
```

Comment:
This function will be called in `BaseTableExecutor` when optimizing configs are changed , but actually `TableRuntimeRefreshExecutor#execute` is the final operation. So if you evaluate pending input in `handleConfigChanged`, then the table will have a duplicate scan plan.

Reply:
Yes, but I didn't find that there is a persistent variable in the `RuntimeTable` that can carry the ConfigChanged event to the `execute` method triggered by the next `TableRuntimeRefreshExecutor`

Reply:
Maybe check the optimizing state in `execute` function is better?

Reply:
> Maybe check the optimizing state in `execute` function is better?

+1, But the key point is how to know in `execute` that the configuration update of the current table has occurred, so as to trigger the `tryEvaluatingPendingInput` method, maybe a persistent field or other solutions are required?

Reply:
before the tableRuntime refresh, we can get the original self-optimizing status

Reply:
> before the tableRuntime refresh, we can get the original self-optimizing status

no, The essence of `handleConfigChanged` is triggered from `refresh`,